src/Symfony/Component/Mailer/DataCollector/MessageDataCollector.php

```diff
@@ -65,4 +67,29 @@ public function getName(): string
     {
         return 'mailer';
     }
+
+    public function path(DataPart $attachment): ?string
+    {
+        $path = tempnam(sys_get_temp_dir(), 'symfony_mailer_attachment_');
+
+        if (false === $path) {
+            return null;
+        }
+
+        $result = file_put_contents($path, $attachment->getBody());
+
+        if (false === $result) {
+            return null;
+        }
+
+        return base64_encode($path);
+    }
+
+    public function filename(DataPart $attachment): string
+    {
+        $reflectionProperty = new ReflectionProperty(DataPart::class, 'filename');
+        $reflectionProperty->setAccessible(true);
+
+        return $reflectionProperty->getValue($attachment);
```

Comment:
Obviously not preferable to use reflection here, but I haven't found any other way to get the filename. Suggestions are welcome!

Reply:
My instinct is to make the filename accessible but we'd still need reflection to support older `symfony/mime` versions...

Reply:
no need to use reflection IMO. Either we bump the min version of `symfony/mime` or we disable this profiler feature for older versions IMO.
Using reflection to access private parts of a different package is a no-go, because that would force us to stop using the semver operator for the requirement of that other package (as the stuff we access would not be covered by semver)

Reply:
Agreed, my [idea below](https://github.com/symfony/symfony/pull/44917#discussion_r779084867) would just disable this feature in symfony/mime < 6.1.